tactics/tactics.ml

```diff
@@ -3074,17 +3074,17 @@ let expand_hyp id = Tacticals.New.tclTRY (unfold_body id) <*> clear [id]
 
  *)
 
-let warn_unused_intro_pattern =
+let warn_unused_intro_pattern env =
   CWarnings.create ~name:"unused-intro-pattern" ~category:"tactics"
          (fun names ->
           strbrk"Unused introduction " ++ str (String.plural (List.length names) "pattern")
           ++ str": " ++ prlist_with_sep spc 
          (Miscprint.pr_intro_pattern 
-	 (fun c -> Printer.pr_econstr (snd (c (Global.env()) Evd.empty)))) names)
+	 (fun c -> Printer.pr_econstr (snd (c env Evd.empty)))) names)
 
-let check_unused_names names =
+let check_unused_names names env =
   if not (List.is_empty names) then
-    warn_unused_intro_pattern names
+    warn_unused_intro_pattern env names
```

Comment:
I don't know how to use this sigma once I add it as a parameter to check_unused_names